diff -uNr a/swf/src/avm1/write.rs b/swf/src/avm1/write.rs
--- a/swf/src/avm1/write.rs	2026-08-17 23:37:41.760730666 -0700
+++ b/swf/src/avm1/write.rs	2026-08-17 23:37:41.944554071 -0700
@@ -11,7 +11,7 @@
 
 impl<W: Write> SwfWriteExt for Writer<W> {
     #[inline]
-    fn as_writer(&mut self) -> &mut impl Write {
+    fn as_writer(&mut self) -> &mut dyn Write {
         &mut self.output
     }
 }
diff -uNr a/swf/src/avm2/write.rs b/swf/src/avm2/write.rs
--- a/swf/src/avm2/write.rs	2026-08-17 23:37:41.943295438 -0700
+++ b/swf/src/avm2/write.rs	2026-08-17 23:37:41.944591904 -0700
@@ -9,7 +9,7 @@
 
 impl<W: Write> SwfWriteExt for Writer<W> {
     #[inline]
-    fn as_writer(&mut self) -> &mut impl Write {
+    fn as_writer(&mut self) -> &mut dyn Write {
         &mut self.output
     }
 }
diff -uNr a/swf/src/write.rs b/swf/src/write.rs
--- a/swf/src/write.rs	2026-08-17 23:37:41.596452884 -0700
+++ b/swf/src/write.rs	2026-08-17 23:37:41.944510612 -0700
@@ -134,7 +134,9 @@
 }
 
 pub trait SwfWriteExt {
-    fn as_writer(&mut self) -> &mut impl Write;
+    fn as_writer(&mut self) -> &mut dyn Write {
+        unimplemented!()
+    }
 
     impl_write_little_endian! {
         write_u8 for u8;
@@ -238,7 +240,7 @@
 
 impl<W: Write> SwfWriteExt for Writer<W> {
     #[inline]
-    fn as_writer(&mut self) -> &mut impl Write {
+    fn as_writer(&mut self) -> &mut dyn Write {
         &mut self.output
     }
 }
