diff -uNr a/swf/src/read.rs b/swf/src/read.rs
--- a/swf/src/read.rs	2026-08-25 00:00:00.000000000 +0000
+++ b/swf/src/read.rs	2026-08-25 00:00:00.000000000 +0000
@@ -1938,6 +1938,7 @@
                 let id = self.read_u16()?;
                 PlaceObjectAction::Replace(id)
             }
+            (false, false) if has_class_name => PlaceObjectAction::PlaceByClass,
             _ => return Err(Error::invalid_data("Invalid PlaceObject type")),
         };
         let matrix = if flags.contains(PlaceFlag::HAS_MATRIX) {
diff -uNr a/swf/src/types.rs b/swf/src/types.rs
--- a/swf/src/types.rs	2026-08-25 00:00:00.000000000 +0000
+++ b/swf/src/types.rs	2026-08-25 00:00:00.000000000 +0000
@@ -360,6 +360,7 @@
     Place(CharacterId),
     Modify,
     Replace(CharacterId),
+    PlaceByClass,
 }
 
 #[derive(Clone, Debug, PartialEq)]
diff -uNr a/swf/src/write.rs b/swf/src/write.rs
--- a/swf/src/write.rs	2026-08-25 00:00:00.000000000 +0000
+++ b/swf/src/write.rs	2026-08-25 00:00:00.000000000 +0000
@@ -1708,6 +1708,7 @@
                 PlaceObjectAction::Place(character_id)
                 | PlaceObjectAction::Replace(character_id) => writer.write_u16(character_id)?,
                 PlaceObjectAction::Modify => (),
+                PlaceObjectAction::PlaceByClass => (),
             }
             if let Some(ref matrix) = place_object.matrix {
                 writer.write_matrix(matrix)?;
